### PR TITLE
Fix configure_containerd.sh script idempotency

### LINKED ISSUE #471: Fix Idempotency of configure_containerd.sh script
"Add option to use a custom container registry in containerd" https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/432 feature introduced a script [configure_containerd.sh](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/files/bin/configure_containerd.sh). This script injects `containerd` registry host and cert files into cluster-template.yaml. 

This injection is not an idempotent operation. This may cause the  `containerd` registry host and cert files could be added on each script execution.

Add a check which evaluates whether the containerd file is already present in the cluster-template.yaml and make `configure_containerd.sh` idempotent.

### PR DESCRIPTION
**Pull request by matofeder**
Script configure_containerd.sh did not evaluate whether the containerd file is already present in the cluster-template.yaml. Hence, the file could be added on each script execution.

This commit adds a check which evaluates whether the containerd file is already present in the cluster-template.yaml. This makes configure_containerd.sh idempotent.

Closes #471

### CODE REVIEW

**Review by fdobrovolny — APPROVED** [review]
LGTM

File: terraform/files/bin/configure_containerd.sh [hunk 1]
```diff
@@ -57,10 +57,23 @@ for path in "${paths[@]}"; do
...
-
-      yq 'select(.kind == "KubeadmControlPlane").spec.kubeadmConfigSpec.files += [load("file_tmp")]' -i "$1"
-      yq 'select(.kind == "KubeadmConfigTemplate").spec.template.spec.files += [load("file_tmp")]' -i "$1"
-
+      # Evaluate whether the file is already present in the cluster-template.yaml.
+      # YAML key `files` is not mandatory therefore it should be added as an empty array to ensure that the whole evaluation will work as expected,
+      # see related YQ docs: https://mikefarah.gitbook.io/yq/operators/alternative-default-value#update-or-create-entity-does-not-exist
+      file_cp_exist=$(yq 'select(.kind == "KubeadmControlPlane").spec.kubeadmConfigSpec | (.files // (.files = []))[] | select(.path == env(destination_path) + env(file_name))' "$1")
+      if test -z "$file_cp_exist"; then
+        echo "Adding $file_name to the in KubeadmControlPlane files"
```
Comment (chess-knight) [inline terraform/files/bin/configure_containerd.sh:65]:
```suggestion
        echo "Adding $file_name to the KubeadmControlPlane files"
```

File: terraform/files/bin/configure_containerd.sh [hunk 2]
```diff
@@ -57,10 +57,23 @@ for path in "${paths[@]}"; do
...
+        yq 'select(.kind == "KubeadmControlPlane").spec.kubeadmConfigSpec.files += [load("file_tmp")]' -i "$1"
+      else
+        echo "$file_name is already defined in KubeadmControlPlane files"
+      fi
+      file_ct_exist=$(yq 'select(.kind == "KubeadmConfigTemplate").spec.template.spec | (.files // (.files = []))[] | select(.path == env(destination_path) + env(file_name))' "$1")
+      if test -z "$file_ct_exist"; then
+        echo "Adding $file_name to the in KubeadmConfigTemplate files"
```
Comment (chess-knight) [inline terraform/files/bin/configure_containerd.sh:72]:
```suggestion
        echo "Adding $file_name to the KubeadmConfigTemplate files"
```

**Review by chess-knight — APPROVED** [review]
Overall LGTM and tested with a second run of `create_cluster.sh`.
Just small typos

**Pull request merged**